### PR TITLE
sql, geo: Fix upper case geohash parsing and allow NULL arguments

### LINKED ISSUE #75537: spatial: difference in st_box2dfromgeohash result
Running on Postgis:
```
postgres=# select st_box2dfromgeohash('F'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D ;
 st_box2dfromgeohash
---------------------
 BOX(-90 45,-45 90)
```

But on CockroachDB:
```
root@:26257/defaultdb> select st_box2dfromgeohash('F'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D ;
  st_box2dfromgeohash
-----------------------
  NULL
```

### PR DESCRIPTION
**Pull request by RichardJCai**
Release note (sql change): ST_Box2DFromGeoHash now accepts NULL arguments,
the precision being NULL is the same as no precision being passed in at all.

Upper case characters are now parsed as lower case characters for geohash,
this matches Postgis behaviour.

Resolves https://github.com/cockroachdb/cockroach/issues/75537

### CODE REVIEW

File: pkg/geo/parse.go [hunk 1]
```diff
@@ -239,6 +239,9 @@ func parseGeoHash(g string, precision int) (geohash.Box, error) {
 		return geohash.Box{}, pgerror.Newf(pgcode.InvalidParameterValue, "length of GeoHash must be greater than 0")
 	}
 
+	// In PostGis, upper case letters are parsed as lower case letters.
```
Comment (otan) [inline pkg/geo/parse.go:242]: nit: capitalise PostGIS as PostGIS
maybe use the term "case insensitive"

**Review by otan — APPROVED** [review]

File: pkg/geo/parse.go (hunk 1 above)
Comment (RichardJCai) [reply pkg/geo/parse.go]: Done.

File: pkg/geo/parse_test.go [hunk 2]
```diff
@@ -642,6 +642,17 @@ func TestParseHash(t *testing.T) {
 				Max: -120.9375,
 			},
 		}},
+		// In PostGIS, upper case characters are parsed as lower case characters.
```
Comment (otan) [inline pkg/geo/parse_test.go:645]: nit: use term "case sensitive"

**Review by otan — APPROVED** [review]

File: pkg/geo/parse_test.go (hunk 2 above)
Comment (RichardJCai) [reply pkg/geo/parse_test.go:645]: case-insensitive right? updated.

**Comment by RichardJCai** [comment]
Thanks!
bors r=otan

**Pull request merged**